src/header/js/header-shopping-cart.js

```diff
@@ -0,0 +1,70 @@
+import {$assign as $, $dispatch, $replaceAll, $renderSvgOrImg} from '../../shared/js/shared';
+import {$cart} from '../../shared/js/iconPaths';
+
+const prefix = 'esri-header-shopping-cart';
+
+export default () => {
+	const $target = $('div', {class: prefix});
+
+	$target.addEventListener('click', (event) => {
+		$dispatch($target, 'header:click:shoppingCart', {event});
+	});
+
+	/* Shopping Cart: Control
+	/* ====================================================================== */
+
+	const $cartItems = $('div',
+		{class: `${prefix}--items`, id: `${prefix}--items`}
+	);
+
+	$target.addEventListener('header:update:cart', ({detail}) => {
+		const $control = $('div',
+			{
+				class: `${prefix}--icon`, id: `${prefix}--icon`,
+				aria: {expanded: false, controls: `${prefix}--content`}
+			}, $renderSvgOrImg({imgDef: $cart.md, imgClass: `${prefix}--image`, id: `${prefix}--image`, $targetElm: $control})
+		);
+
+		$($target,
+			$control,
+			$cartItems
+		);
+		changeCartCount(detail.items);
+	});
+
+	/* Shopping Cart: Target
+	/* ====================================================================== */
+
+	$target.addEventListener('header:shoppingcart:add', ({detail}) => {
+		changeCartCount(detail, true);
+	});
+
+	$target.addEventListener('header:shoppingcart:remove', ({detail}) => {
+		changeCartCount(-detail, true);
+	});
+
+	const changeCartCount = (inc, animate) => {
+		let currCount = parseInt($cartItems.innerHTML);
+		currCount = (isNaN(currCount) || currCount < 0) ? 0 : currCount;
+
+		const cartCount = currCount + parseInt(inc);
+		$cartItems.innerHTML = cartCount;
+
+		if (cartCount > 0) {
+			$target.setAttribute('data-cart-empty', 'false');
+			if (animate) {
+				$cartItems.setAttribute('data-cart-updated', 'true');
+				setTimeout(() => {
+					$cartItems.setAttribute('data-cart-updated', 'false');
+				}, 1000);
```

Comment:
What is this delay for?

Reply:
because when an item is added to the cart it animates the bubble bigger for a second then it waits a second and then animates back to it's original state

Reply:
Couldn't you set the animation time to`2s` and place 4 steps within `@keyframes pulse`? Like  0%, 50%, 75%, 100%.

Reply:
@dlewis-esri  I was trying to get the same animation effect with just the keyframes like @gschnall  suggested, but it doesn't have the same transitions, and it became a bit choppy. I am ok with this approach.

Reply:
We talked this through and the approach with the set timeout is the best path as we need the data attribute added to show the animation in when an item is added to the cart but then we need the data attribute removed so that if a second item is added to the cart we can trigger the animation again.  If we do the animate in and animate out on the same class we still have to remove the attribute with a set timeout anyways to allow additional items to be added.